Towny/src/main/java/com/palmergames/bukkit/towny/object/economy/transaction/Transaction.java

```diff
@@ -0,0 +1,68 @@
+package com.palmergames.bukkit.towny.object.economy.transaction;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.Nullable;
+
+import com.palmergames.bukkit.towny.object.economy.Account;
+
+public class Transaction {
+	private final TransactionType type;
+	private final Account receivingAccount;
+	private final Account sendingAccount;
+	private final double amount;
+
+	public Transaction(TransactionBuilder builder) {
+		this.type = builder.type; 
+		this.receivingAccount = builder.receivingAccount;
+		this.sendingAccount = builder.sendingAccount;
+		this.amount = builder.amount;
+	}
```

Comment:
I would make this a normal constructor instead of requiring the use of the builder

```suggestion
	@Contract("_, null, null, _ -> fail")
	public Transaction(@NotNull TransactionType type, Account receivingAccount, Account sendingAccount, double amount) {
		Preconditions.checkNotNull(type, "type");
		Preconditions.checkState(receivingAccount != null || sendingAccount != null, "receivingAccount and sendingAccount cannot both be null");
		Preconditions.checkState(amount > 0, "amount must be greater than 0");
		this.type = type;
		this.receivingAccount = receivingAccount;
		this.sendingAccount = sendingAccount;
		this.amount = amount;
	}
```